tutorials/basic-topics/sorting/merge-sort.md

```diff
@@ -11,4 +11,519 @@ keywords:
+
+Then we can clearly see that merge sort is performed like this way. 
+
+![image](https://miro.medium.com/max/600/1*opwN0BhtH4zvPF697fPlow.gif)
+
+## Algorithm 
+
+### Divide ( Dividing is our first step)
+
+- Calculate the mid point by checking if left index is less than the right index , if yes divide the array . 
+- Now continue dividing the array until `left index < right index` becomes false , that is until division is not possible.
+     
+### Conquer
+
+- After dividing the array into smallest units, start merging the elements again by comparing them .
+
+- We need to compare and merge starting from the last splits or last smallest units. So **Recursion** needs to done here.
+
+### Merge 
+
+- Since each half is already sorted so we need to just sort between 2 halves to **combine/merge** then to make a bigger sorted array
+   
+## [Leetcode Problem 912 Sort an Array](https://leetcode.com/problems/sort-an-array/)
+
+**Problem Statement** 
+
+```
+Given an array of integers nums, sort the array in ascending order and return it.
+
+You must solve the problem without using any built-in functions in O(nlog(n)) time complexity and with the smallest space complexity possible.
+
+```
+
+**Test Cases**
+    
+*Test case 1* :
+     
+```
+Input: nums = [5,2,3,1]
+Output: [1,2,3,5]
+Explanation: 
+After sorting the array, the positions of some numbers are not changed (for example, 2 and 3), while the positions of other numbers are changed (for example, 1 and 5).
+        
+```
+**Constraints as given in the problem:**
+
+$1 <= nums.length <= 5 * 10^4
+-5 * 10^4 <= nums[i] <= 5 * 10^4$ 
+
+*Here `insertion sort` , `quick sort` , `selection sort`  and all other sorts in $O(n^2)$ will give **TLE** i.e Time Limit Exceeded*
+
+## STL Approach  
+Note: Not preferred here in this problem , but still you will get a AC in Leetcode
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+
+```cpp
+void stableStlSort(vector<int>& nums) {
+stable_sort(nums.begin(), nums.end());
+}
+```
+</TabItem>
+
+<TabItem value="py" label="Python">
+
+```py
+def sortArray(self, nums):
+    return sorted(nums)
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+public int[] sortArray(int[] nums) {
+    Arrays.sort(nums);
+    return nums;
+}
+```
+</TabItem>
+</Tabs>
+
+
+## Top Down Approach
+
+### Algorithm 
+    
+```
+MergeSort(array[], left,  right)
+If right > left
+
+Find the middle point to divide the array into two halves: 
+middle = left + (right – left)/2
+Call mergeSort again for first half for further dividing:   
+Call mergeSort(array, left, middle)
+Call mergeSort again  for second half for further dividing:
+Call mergeSort(array, middle+ 1, right)
+Merge the two halves sorted in steps 2 and 3:
+Call merge(arrary, left, middle, right)
+```
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+
+```cpp
+// call mergeSort(nums, 0, nums.size() - 1) to sort 
+
+void merge(vector<int>& nums, int l, int m, int r){
```

Comment:
Format the intentation in IDE and add it here.

Reply:
Intentation has to be corrected. Use IDE instead of formatting in markdown editor. 

![image](https://user-images.githubusercontent.com/10242660/193452850-4310c93a-5275-413c-be62-deebdfeef48f.png)